addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRuleUnitTest.java

```diff
@@ -121,4 +129,225 @@ public void replaceMultipleElementsAndRemoveHeader() {
+    }
+
+    @ParameterizedTest
+    @EnumSource(
+            value = NanoHTTPD.Response.Status.class,
+            names = {"OK", "BAD_REQUEST"})
+    public void shouldAlertWhenLocalFileIncludedInResponse(NanoHTTPD.Response.Status status)
+            throws Exception {
+        // Given
+        String test = "/test";
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
+                        consumeBody(session);
+                        String response = "root:*:0:0:System Administrator:/var/root:/bin/sh";
+                        return newFixedLengthResponse(status, NanoHTTPD.MIME_PLAINTEXT, response);
+                    }
+                });
+        HttpMessage msg = this.getHttpMessage(test);
+        msg.setRequestBody("<?xml version=\"1.0\"?><comment><text>test</text></comment>");
+        msg.getRequestHeader().setMethod("POST");
+        msg.getRequestHeader().setHeader("Content-Type", "application/xml");
+        rule.init(msg, parent);
+
+        // When
+        rule.localFileInclusionAttack(msg);
+
+        // Then
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("root:*:0:0"));
+        assertThat(
+                alertsRaised.get(0).getAttack(),
+                equalTo(
+                        MessageFormat.format(XxeScanRule.ATTACK_HEADER, "file:///etc/passwd")
+                                + XxeScanRule.ATTACK_BODY));
+        assertThat(alertsRaised.get(0).getRisk(), equalTo(Alert.RISK_HIGH));
+        assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
+    }
+
+    @Test
+    public void shouldStopLocalFileInclusionRuleIfScanIsStopped() throws Exception {
+        // Given
+        String test = "/test";
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
+                        consumeBody(session);
+                        String response =
+                                "<foo>root:*:0:0:System Administrator:/var/root:/bin/sh</foo>";
+                        return newFixedLengthResponse(
+                                NanoHTTPD.Response.Status.OK, NanoHTTPD.MIME_PLAINTEXT, response);
+                    }
+                });
+        HttpMessage msg = this.getHttpMessage(test);
+        msg.setRequestBody("<?xml version=\"1.0\"?><comment><text>test</text></comment>");
+        msg.getRequestHeader().setMethod("POST");
+        msg.getRequestHeader().setHeader("Content-Type", "application/xml");
+        parent.stop();
+        rule.init(msg, parent);
+
+        // When
+        rule.localFileInclusionAttack(msg);
+
+        // Then
+        assertThat(countMessagesSent, equalTo(1));
+    }
```

Comment:
This test is an interesting idea, however, I'm not sure it'll behave properly if When is `rule.scan` and `parent.stop()` was issued in Given.
@thc202? @psiinon?

Reply:
This test did started failing after using rule.scan(), but not sure if it is for the same reason you had in mind. The reason it started failing for me was: since `localFileInclusionAttack` and `localFileReflectionAttack` has been moved to its own methods, `return`, when `parent.stop()` is true, exits only that method and not the scan itself. So I have added another check within the `scan()` method between attacks to stop scanning when `parent.stop()` is invoked. This fixed the test.

Reply:
I'm not sure we should add it as is, if the scanner is already stopped the scan rule would not have been called (it could happen immediately after but it's less likely than when actually running), better to test that the stop is respected by stopping while running (i.e. in the several loops that it might have).

Reply:
This is probably something that can be added as part of the common tests.

Reply:
For the time being lets just leave the stop testing out for now. It would be nice to get this done/in. Stop testing will require further unrelated changes.

Reply:
Sure. Will remove this test.

Reply:
This test has been removed.